### PR TITLE
add more magit-display-buffer-function options

### PR DESCRIPTION
**Pull request by kyleam**
Thoughts?  Other variants to consider including?

### CODE REVIEW

File: lisp/magit-mode.el [hunk 1]
```diff
@@ -99,9 +99,12 @@ inside your function."
 All Magit buffers (buffers whose major-modes derive from
 `magit-mode') are displayed using `magit-display-buffer',
 which in turn uses the function specified here."
-  :package-version '(magit . "2.3.0")
+  :package-version '(magit . "2.7.1")
```
Comment (npostavs) [inline lisp/magit-mode.el]: Shouldn't this be left unchanged, since the default value is not changing?

Comment (kyleam) [inline lisp/magit-mode.el]: Doh, yep.

**Comment by kyleam** [comment]
@Silex @jimeh @fxleblanc, if any of you have time, it'd be great to
get your feedback.  @Silex, `magit-display-buffer-fullframe-status-v1`
is the option for the full frame status buffer (#1953).  @jimeh
@fxleblanc, `magit-display-buffer-same-window-except-diff-v1` should
have a similar behavior to what was requested in #2541.

**Comment by jimeh** [comment]
@kyleam The `magit-display-buffer-same-window-except-diff-v1` display function does exactly what I want, thanks :)

**Comment by kyleam** [comment]
@jimeh Thanks for testing it out.

Applied  (fad19f5de2c964077758f4015efc856b5d64ab14).

**Comment by tarsius** [comment]
I've it turns out that one of these functions still needs some tweaking *before* the next release, then we should just do it, instead of adding a new version. Just saying :-)

> Other variants to consider including?

Personally I haven't gone with fullframe-status yet, because I am so used to having the status (or log) buffer at the bottom but when a second window is created to display another Magit buffer, then *that* buffer is displayed at the bottom. So something that uses `display-buffer-above-selected` (doesn't exist yet) would be nice.

**Comment by kyleam** [comment]
> I've it turns out that one of these functions still needs some tweaking before the next release, then we should just do it, instead of adding a new version. Just saying :-)

Right, my thinking too.

> Personally I haven't gone with fullframe-status yet, because I am so used to having the status (or log) buffer at the bottom but when a second window is created to display another Magit buffer, then that buffer is displayed at the bottom. So something that uses display-buffer-above-selected (doesn't exist yet) would be nice.

Interesting.  I'll look into that.

**Comment by kyleam** [comment]
`km/display-topleft` adds a function that I hope gets close to the behavior you want.

**Comment by tarsius** [comment]
That works very well. Thanks!

(Not so well when using `magit-mode-quit-window` as `magit-bury-buffer-function`, so I went back to the default `magit-restore-window-configuration` for now.)

**Comment by kyleam** [comment]
Great.  I've added the function, with a few tweaks to make it play better
with `magit-mode-bury-buffer`.

**Pull request closed without merging**